### DISCUSSION TITLE
Theia loads indefinitely after update to 0.9.0-RC01

### DISCUSSION THREAD

**Discussion by LiamWalsh98** [eclipse-glsp/glsp#469] [Q&A]
Hello,

I recently attempted to update to 0.9.0-RC01. I went through and checked all the breaking changes. The server update went without a hitch, but the client update unfortunately did not. Theia seems to now refuse to launch. It displays the startup loading gif indefinitely and the JS Console outputs the following:

```
TypeError: Class constructor NotificationManager cannot be invoked without 'new'
    at new GLSPNotificationManager (vendors-node_modules_eclipse-glsp_theia-integration_lib_browser_index_js.bundle.js:16907)
    at _createInstance (bundle.js:78397)
    at Object.resolveInstance (bundle.js:78417)
    at bundle.js:78512
    at Object.resolve (bundle.js:78536)
    at bundle.js:77555
    at Container.__webpack_modules__.../node_modules/inversify/lib/container/container.js.Container._get (bundle.js:77546)
    at Container.__webpack_modules__.../node_modules/inversify/lib/container/container.js.Container.get (bundle.js:77463)
    at Binding.dynamicValue (bundle.js:78770)
    at bundle.js:78503
```
None of the lines in question link to places in the source.

After a little bit of research I've found that similar errors have been fixed by updating ES5 to ES6, but this didn't seem to change anything. I would guess that this is caused by something that I've missed in reviewing the breaking changes. However, if anyone recognizes this particular error, any help would be appreciated.

Thanks

**Comment by tortmayr** [comment]
Hi, @LiamWalsh98 

we have just released GLSP 0.9.0. Could you please try to upgrade your project to this version and let us know if the issues continue?
FYI I'm working on a migration guide right now that should make it easier for adopters to migrate from GLSP 0.8.0 to 0.9.0.
As you already noticed we have also updated the npm packages to ES6 in this release (to be compatible with the latest Theia versions), so you should definitly also upgrade your custom packages to ES6.

**Reply by LiamWalsh98** [reply]
I'll give it a try now! Thanks for letting me know. And I'll keep my eyes open for the migration guide if I can't manage to make it work on my own. Thank you

**Reply by LiamWalsh98** [reply]
By the way, I wonder, is there a rough ETA on the migration guide? Thanks

**Reply by tortmayr** [reply]
I hope to get it ready early next week, but I can't make any promises. You can subscribe to the corresponding GH issue to retrieve updates as soon as the PR is ready (#478).

**Reply by LiamWalsh98** [reply]
No worries at all, sounds good! Thanks for the update
